### PR TITLE
Upgrade pyo3 to 0.15

### PR DESCRIPTION
**Pull request by messense**
This would require dropping Python 3.5 support.

### CODE REVIEW

File: bindings/python/Cargo.toml [hunk 1]
```diff
@@ -29,9 +29,3 @@ tempfile = "3.1"
 
 [features]
 default = ["pyo3/extension-module"]
-
-[target.x86_64-apple-darwin]
```
Comment (messense) [inline bindings/python/Cargo.toml:33]: This is simply wrong place to put this.

```
warning: unused manifest key: target.x86_64-apple-darwin.rustflags
```

**Comment by n1t0** [comment]
Thank you very much for your help with this @messense! I didn't have much time lately but I plan to move forward with this next week probably.

File: bindings/python/Cargo.toml [hunk 2]
```diff
@@ -14,8 +14,8 @@ serde = { version = "1.0", features = [ "rc", "derive" ]}
 serde_json = "1.0"
 libc = "0.2"
 env_logger = "0.7.1"
-pyo3 = "0.12"
-numpy = "0.12"
+pyo3 = "0.15.0"
+numpy = "0.15.0"
```
Comment (h-vetinari) [inline bindings/python/Cargo.toml:18]: numpy does not need to move in sync with pyo3; there also does not [exist](https://github.com/numpy/numpy/tags?after=v1.0) a numpy 0.15...

Comment (messense) [reply bindings/python/Cargo.toml:18]: It needs to. And it's [rust-numpy](https://github.com/PyO3/rust-numpy).

**Comment by messense** [comment]
I'm no long interested in pushing this forward given the lack of review for a year now. Feel free to take over.

File: bindings/python/Cargo.toml (hunk 2 above)
Comment (h-vetinari) [reply bindings/python/Cargo.toml:18]: OK, makes sense, thanks.

**Comment by h-vetinari** [comment]
> I'm no long interested in pushing this forward given the lack of review for a year now. Feel free to take over.

Sorry to hear. I ran into needing this while packaging tokenizers for conda-forge (py3.10 builds fail in cross-compilation without pyo3 0.15)

**Comment by Narsil** [comment]
@messense @h-vetinari Sorry it took so long to review, @n1t0 is hard at work at HF, and simply stopped having time maintaining `tokenizers`.

I am spending some time to maintain it because I love the library (and rust).
Sorry I didn't really look at very old PRs when I "took over".

Feel free to ping regularly if this ever happens again, GH notifications are sometimes not working too well (or dismissed by error).

Thanks @h-vetinari for taking this over ! 

Cheers !

**Comment by h-vetinari** [comment]
If @messense is not too demotivated to take this to completion, this would surely go faster than I'd ever pull it off as a pretty complete newbie. No expectation, but just saying. 😅

**Pull request closed without merging**